lua/astrocommunity/pack/typescript-tools/init.lua

```diff
@@ -0,0 +1,120 @@
+    opts = function(_, opts)
+      if opts.ensure_installed ~= "all" then
+        opts.ensure_installed = utils.list_insert_unique(opts.ensure_installed, { "javascript", "typescript", "tsx" })
+      end
+    end,
+  },
+  {
+    "jay-babu/mason-null-ls.nvim",
+
+    opts = function(_, opts)
+      opts.ensure_installed = utils.list_insert_unique(opts.ensure_installed, { "prettierd", "eslint_d" })
+      if not opts.handlers then opts.handlers = {} end
+
+      local has_prettier = function(util)
+        return check_json_key_exists(vim.fn.getcwd() .. "/package.json", "prettier")
+          or util.root_has_file ".prettierrc"
+          or util.root_has_file ".prettierrc.json"
+          or util.root_has_file ".prettierrc.yml"
+          or util.root_has_file ".prettierrc.yaml"
+          or util.root_has_file ".prettierrc.json5"
+          or util.root_has_file ".prettierrc.js"
+          or util.root_has_file ".prettierrc.cjs"
+          or util.root_has_file "prettier.config.js"
+          or util.root_has_file "prettier.config.cjs"
+          or util.root_has_file ".prettierrc.toml"
+      end
+
+      local has_eslint = function(util)
+        return util.root_has_file ".eslintrc.js"
+          or util.root_has_file ".eslintrc.cjs"
+          or util.root_has_file ".eslintrc.yaml"
+          or util.root_has_file ".eslintrc.yml"
+          or util.root_has_file ".eslintrc.json"
+          or check_json_key_exists(vim.fn.getcwd() .. "/package.json", "eslintConfig")
+      end
+
+      opts.handlers.prettierd = function()
+        local null_ls = require "null-ls"
+        null_ls.register(null_ls.builtins.formatting.prettierd.with { condition = has_prettier })
+      end
+
+      opts.handlers.eslint_d = function()
+        local null_ls = require "null-ls"
+        null_ls.register(null_ls.builtins.diagnostics.eslint_d.with { condition = has_eslint })
+        null_ls.register(null_ls.builtins.formatting.eslint_d.with { condition = has_eslint })
+        null_ls.register(null_ls.builtins.code_actions.eslint_d.with { condition = has_eslint })
+      end
+    end,
+  },
+  {
+    "jay-babu/mason-nvim-dap.nvim",
+    opts = function(_, opts) opts.ensure_installed = utils.list_insert_unique(opts.ensure_installed, "js") end,
+  },
+  {
+    "vuki656/package-info.nvim",
+    dependencies = { "MunifTanjim/nui.nvim" },
+    opts = {},
+    event = "BufRead package.json",
+  },
+  {
+    "pmizio/typescript-tools.nvim",
+    dependencies = { "nvim-lua/plenary.nvim", "neovim/nvim-lspconfig" },
+    ft = {
+      "typescript",
+      "typescriptreact",
+      "javascript",
+      "javascriptreact",
+    },
+    opts = {
+      on_attach = function(client)
+        client.server_capabilities.documentFormattingProvider = false
+        client.server_capabilities.documentRangeFormattingProvider = false
+      end,
+      settings = {
+        expose_as_code_action = { "fix_all", "add_missing_imports", "remove_unused" },
+      },
+    },
+  },
```

Comment:
We'll need to skip the setup of tsserver server by astronvim, since typescript-tools.nvim does that for you. 
```lua
init = function() astronvim.lsp.skip_setup = utils.list_insert_unique(astronvim.lsp.skip_setup, "tsserver") end,
```
Should do the trick.
Also for the opts. Wy are you disabling documentFormattingProvider and documentRangeFormattingProvider?
Same goes for the settings.

Reply:
ping @Cretezy